sympy/core/function.py

```diff
@@ -1005,10 +1005,15 @@ def __new__(cls, expr, *variables, **assumptions):
             variables = expr.free_symbols
             if len(variables) != 1:
                 from sympy.utilities.misc import filldedent
-                raise ValueError(filldedent('''
-                    Since there is more than one variable in the
-                    expression, the variable(s) of differentiation
-                    must be supplied to differentiate %s''' % expr))
+                if len(variables) == 0:
+                    raise ValueError(filldedent('''
+                        Since there are no variables in the expression,
+                        %s cannot be differentiated''' % expr))
```

Comment:
It can be differentiated after supplying variable.`t.diff(t)` will work, but `t.diff()` will not.